apps/web/modules/ee/contacts/attributes/components/edit-attribute-modal.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import toast from "react-hot-toast";
+import { useTranslation } from "react-i18next";
+import { TContactAttributeKey } from "@formbricks/types/contact-attribute-key";
+import { getFormattedErrorMessage } from "@/lib/utils/helper";
+import { Button } from "@/modules/ui/components/button";
+import {
+  Dialog,
+  DialogBody,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/modules/ui/components/dialog";
+import { Input } from "@/modules/ui/components/input";
+import { updateContactAttributeKeyAction } from "../actions";
+
+interface EditAttributeModalProps {
+  attribute: TContactAttributeKey;
+  open: boolean;
+  setOpen: (open: boolean) => void;
+}
+
+export function EditAttributeModal({ attribute, open, setOpen }: Readonly<EditAttributeModalProps>) {
+  const { t } = useTranslation();
+  const router = useRouter();
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [formData, setFormData] = useState({
+    name: attribute.name ?? "",
+    description: attribute.description ?? "",
+  });
+
+  const handleUpdate = async () => {
```

Comment:
please handle any errors that might be thrown in this async function using try...catch...finally 🙏